client/src/sessions/password.rs

```diff
@@ -1,14 +1,43 @@
 use crate::sessions::Session;
-use tpm2_rs_base::errors::{TssResult, TssTcsError};
+use tpm2_rs_base::errors::{TpmRcResult, TssResult, TssTcsError};
 use tpm2_rs_base::{
     Tpm2bAuth, Tpm2bNonce, Tpm2bSimple, TpmaSession, TpmiShAuthSession, TpmsAuthCommand,
     TpmsAuthResponse,
 };
 
 /// A password session.
+///
+/// # Usage:
+/// ```
+/// use tpm2_rs_client::sessions::PasswordSession;
+///
+/// let password1 = PasswordSession::new("hello world").unwrap(); // from a string
```

Comment:
I think this actually isn't supported by the code below; but it can be. You'll have to update your template to:

```rs
pub fn new<T: AsRef<[u8]> + ?Sized>(password: T) -> ...
```

This is because a `&str` isn't `Sized`, and the default assumption is that types are `Sized` unless said otherwise.

Reply:
I am not fully sure what you mean by this being not supported, it just ran as part of `cargo test` and was accepted, unless I am missing something

Reply:
It only needs `?Sized` if you also take https://github.com/tpm-rs/tpm-rs/pull/113#discussion_r1800045358

Which it seems like you are not.